CronosPlaySdk/Plugins/CronosPlayUnreal/Source/CronosPlayUnreal/Private/PlayCppSdkBPLibrary.cpp

```diff
@@ -180,3 +201,142 @@ UTexture2D *UPlayCppSdkBPLibrary::GenerateQrCode(FString string) {
+    FHttpModule &httpModule = FHttpModule::Get();
+    TSharedRef<IHttpRequest, ESPMode::ThreadSafe> httprequest =
+        httpModule.CreateRequest();
+    httprequest->SetVerb(TEXT("GET"));
+    httprequest->SetHeader(TEXT("Content-Type"),
+                           TEXT("application/x-www-form-urlencoded"));
+    httprequest->SetHeader(TEXT("User-Agent"), g_user_agent);
+    httprequest->SetURL(tokenuri);
+
+    httprequest->OnProcessRequestComplete().BindLambda(
+
+        [&](FHttpRequestPtr callbackrequest, FHttpResponsePtr callbackresponse,
+            bool connectedSuccessfully) mutable {
+          FString result;
+          FString jsonstring;
+          FCronosNftInfo nftinfo;
+          if (connectedSuccessfully) {
+            FString response = callbackresponse->GetContentAsString();
+            jsonstring = response;
+            TSharedRef<TJsonReader<TCHAR>> jsonreader =
+                TJsonReaderFactory<TCHAR>::Create(jsonstring);
+            TSharedPtr<FJsonObject> jsonobject =
+                MakeShareable(new FJsonObject());
+            FJsonSerializer::Deserialize(jsonreader, jsonobject);
+
+            nftinfo.JsonString = jsonstring;
+            jsonobject->TryGetStringField(TEXT("name"), nftinfo.Name);
+            jsonobject->TryGetStringField(TEXT("description"),
+                                          nftinfo.Description);
+            jsonobject->TryGetStringField(TEXT("image"), nftinfo.Image);
+            nftinfo.ImageUrl = convert_ipfs_url(nftinfo.Image);
+            result = TEXT("");
+          } else {
+            switch (callbackrequest->GetStatus()) {
+            case EHttpRequestStatus::Failed_ConnectionError:
+              result = TEXT("Connection failed.");
+            default:
+              result = TEXT("Request failed.");
+            }
+          }
+
+          AsyncTask(ENamedThreads::GameThread, [Out, nftinfo, result]() {
+            Out.ExecuteIfBound(nftinfo, result);
+          });
+        });
+
+    httprequest->ProcessRequest();
+  });
+}
+
+void UPlayCppSdkBPLibrary::GetNftImageFromUrl(FGetNftImageFromUrl Out,
+                                              FString imageurl) {
+  AsyncTask(ENamedThreads::AnyHiPriThreadNormalTask, [Out, imageurl]() {
+    FHttpModule &httpModule = FHttpModule::Get();
+    TSharedRef<IHttpRequest, ESPMode::ThreadSafe> httprequest =
+        httpModule.CreateRequest();
+    httprequest->SetVerb(TEXT("GET"));
+    httprequest->SetHeader(TEXT("User-Agent"), g_user_agent);
+    httprequest->SetURL(imageurl);
+
+    httprequest->OnProcessRequestComplete().BindLambda(
+        [&](FHttpRequestPtr callbackrequest, FHttpResponsePtr callbackresponse,
+            bool connectedSuccessfully) mutable {
+          UTexture2D *nftimage = NULL;
+          FString result;
+          if (connectedSuccessfully) {
+
+            nftimage = UKismetRenderingLibrary::ImportBufferAsTexture2D(
+                NULL, callbackresponse->GetContent());
```

Comment:
I'd still probably caution against loading arbitrary data from the internet

Reply:
i'll add caution